### ISSUE TITLE
Can it be easy to support R=256, S=256, A=256 ? Even with less layers like 20.

### ISSUE THREAD

**Issue by yxlijinzhu** [NVIDIA/nv-wavenet#26]
Great to see this implementation. We're trying to leverage GPU to do the wave.net inference, while our model's R=256, and we want to see if it is easy to support it and archive real time for at least 16k sample rate.

**Comment by PetrochukM** [comment]
Ditto above ^. In addition, allow a dilation rate of 3 to replicate Tacotron 2. Happy to help make this happen!

**Comment by PetrochukM** [comment]
@rafaelvalle What do you think?

**Comment by rafaelvalle** [comment]
I think we should do ablation studies and post samples to evaluate if these changes in dilation rate and model size are necessary.

The Tacotron 2 paper gives:
    4.58 MOS for professionally recorded speech;
    4.481 for the network with 2 cycles of 6 layers and receptive field of 253 samples (10.5 ms). 

They also mention that:
    "we find that our model can generate high quality audio using as few as 12 layers with a receptive 
      field of 10.5 ms, compared to 30 layers and 256 ms in the baseline model. These results confirm 
      the observations in [9] that a large receptive field size is not an essential factor for audio 
      quality."

Knowing these things, it can be that the numerical difference in MOS is not perceptually meaningful and we're better off choosing a smaller model with faster inference. Unfortunately Google hasn't shared samples from the other models... 

Having said that, allowing for a dilation rate of 3 might be easier than supporting largers channels.
@BrianPharris or @RPrenger can you comment on enabling a dilation rate of 3 on nv-wavenet?

**Comment by PetrochukM** [comment]
Sorry! To clarify, I wrote that post a bit quickly, by “dilation rate” I mean kernel size.

**Comment by rafaelvalle** [comment]
We're evaluating supporting larger channel counts and kernel sizes. As you pointed out, WaveNet, Tacotron 2 are ambiguous regarding their setup. This includes their use in Google Duplex.

Remember though that we want to evaluate WaveNet as a decoder independent of the model that is being used to generate mel-spectrograms.

**Comment by PetrochukM** [comment]
@rafaelvalle For sure. Thanks for your work! This has been super helpful for us. I hope my feedback and pull requests have helped a bit in return.

**Comment by rafaelvalle** [comment]
@PetrochukM yes, the feedback and pull requests have been helpful, thank you.
Regarding Taco 2, I think there has been doubt about exactly what the decoder architecture is, more specifically, the number of LSTMs in the decoder. 
For example, does the Location Sensitive Attention layer have an LSTM in addition to the 2 LSTMs of the decoder?

**Comment by PetrochukM** [comment]
No, the Location Sensitive Attention Layer of Tacotron 2 does not have an LSTM.

Asked them: "Location-Sensitive Attention requires a query vector (e.g. the down arrow from the 2 LSTM layers). Do you use the output of second LSTM layer as a query vector? Or do you use the hidden state of the LSTM layers as a query vector?"

Answer: "we use the output of the first lstm layer as the query vector"

**Comment by rafaelvalle** [comment]
@PetrochukM, thanks for confirming. if you want, let's move this discussion to https://github.com/NVIDIA/tacotron2

**Comment by PetrochukM** [comment]
Ping! Anything going on with regards to this?
